features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Scopes/EditScope.jsx

```diff
@@ -0,0 +1,231 @@
+        marginLeft: theme.spacing.unit * 5,
+    },
+    topics: {
+        marginTop: theme.spacing.unit * 10,
+    },
+    headline: {
+        paddingTop: theme.spacing.unit * 1.5,
+        paddingLeft: theme.spacing.unit * 2.5,
+    },
+});
+
+/**
+ * Create new scopes for an API
+ * @class CreateScope
+ * @extends {Component}
+ */
+class EditScope extends React.Component {
+    constructor(props) {
+        super(props);
+        // this.api = new Api();
+        this.api_uuid = props.match.params.api_uuid;
+        const { api } = this.props;
+        this.state = {
+            apiScope: api.scopes.filter((scope) => {
+                return scope.name === this.props.location.state.scopeName;
+            })[0],
+        };
+        this.addScope = this.addScope.bind(this);
+        this.handleInputs = this.handleInputs.bind(this);
+    }
+
+    /**
+     * Add new scope
+     * @memberof Scopes
+     */
+    addScope() {
+        const restApi = new Api();
+        const updatedScope = this.state.apiScope;
+        const { intl } = this.props;
+        const { api } = this.props;
+        api.scopes = api.scopes.map((scope) => {
+            if (scope.name === updatedScope.name) {
+                return updatedScope;
+            } else {
+                return scope;
+            }
+        });
+        const { history } = this.props;
+        // eslint-disable-next-line no-underscore-dangle
+        const promisedApiUpdate = restApi.update(api._data);
+        promisedApiUpdate.then((response) => {
+            if (response.status !== 200) {
+                Alert.info(intl.formatMessage({
+                    id: 'Apis.Details.Scopes.CreateScope.something.went.wrong.while.updating.the.scope',
+                    defaultMessage: 'Something went wrong while updating the scope',
+                }));
+                return;
+            }
+            Alert.info(intl.formatMessage({
+                id: 'Apis.Details.Scopes.CreateScope.scope.updated.successfully',
+                defaultMessage: 'Scope updated successfully',
+            }));
+            const redirectURL = '/apis/' + api.id + '/scopes/';
+            history.push(redirectURL);
+        });
+    }
+
+    /**
+     * Handle api scope addition event
+     * @param {any} event Button Click event
+     * @memberof Scopes
+     */
+    handleInputs(event) {
+        if (Array.isArray(event)) {
+            const { apiScope } = this.state;
+            apiScope.bindings.values = event;
+            this.setState({
+                apiScope,
+            });
+        } else {
+            const input = event.target;
+            const { apiScope } = this.state;
+            apiScope[input.id] = input.value;
+            this.setState({
+                apiScope,
+            });
+        }
+    }
+
+    render() {
+        const { classes } = this.props;
+        const url = `/apis/${this.props.api.id}/scopes`;
+        return (
```

Comment:
can destructure `api` as well. ex ; `const { classes, api } = this.props`;

Reply:
What ??